nesta_ds_utils/plotting.py

```diff
@@ -0,0 +1,114 @@
+"""
+Module containing utils for styling and exporting figures using Altair.
+"""
+
+import altair_saver as alt_saver
+from selenium import webdriver
+from webdriver_manager.chrome import ChromeDriverManager
+from selenium.webdriver.chrome.webdriver import WebDriver
+from selenium.webdriver.chrome.options import Options
+
+import os
+from typing import Iterator
+from pathlib import Path
+from nesta_ds_utils import file_ops
+
+
+def _google_chrome_driver_setup() -> WebDriver:
+    """Set up the driver to save figures"""
+    chrome_options = Options()
+    chrome_options.add_argument("--headless")
+    driver = webdriver.Chrome(
+        ChromeDriverManager().install(), chrome_options=chrome_options
+    )
+    return driver
+
+
+def _create_paths(path: os.PathLike, filetypes: Iterator[list]):
+    """Checks if the paths exist and if not creates them.
+
+    Args:
+        path (os.PathLike: Path to save figure.
+        filetypes (Iterator[list]): List of file types to save figure.
+    """
+    for filetype in filetypes:
+        os.makedirs(f"{path}/{filetype}", exist_ok=True)
+
+
+def _save_png(fig, path: os.PathLike, name: str, driver: WebDriver):
+    """Save altair chart as a  raster png file.
+
+    Args:
+        fig: Altair chart.
+        path (os.PathLike): Path where to save the figure.
+        name (str): Name of figure.
+        driver (WebDriver): webdriver to use for saving.
+    """
+    alt_saver.save(
+        fig,
+        f"{path}/png/{name}.png",
+        method="selenium",
+        webdriver=driver,
+        scale_factor=5,
+    )
+
+
+def _save_html(fig, path: os.PathLike, name: str):
+    """Save altair chart as a html file.
+
+    Args:
+        fig: Altair chart.
+        path (os.PathLike): Path where to save the figure.
+        name (str): Name of figure.
+    """
+    fig.save(f"{path}/html/{name}.html")
+
+
+def _save_svg(fig, path: os.PathLike, name: str, driver: WebDriver):
+    """Save altair chart as vector svg file.
+
+    Args:
+        fig: Altair chart.
+        path (os.PathLike): Path where to save the figure.
+        name (str): Name of figure.
+        driver (WebDriver): webdriver to use for saving.
+    """
+    alt_saver.save(fig, f"{path}/svg/{name}.svg", method="selenium", webdriver=driver)
+
+
+def save(
+    fig,
+    name: str,
+    path: os.PathLike = None,
+    driver: WebDriver = None,
+    filetypes: Iterator[list] = None,
```

Comment:
Are people usually saving figures as multiple types or do they usually select one of the options? May make sense to allow this to be a string or a list, because I would think people usually only save the image once, but allow a list in case they want multiple? Similarly, should it default to 'png' or one of the other file types if that is more commonly used rather than None?